libs/language-server/src/test/validation/checks/pipe-definition.spec.ts

```diff
@@ -0,0 +1,216 @@
+      );
+
+      expect(validationAcceptorMock).toHaveBeenCalledTimes(0);
+    });
+
+    it('error on unsupported pipe between Blocktypes', async () => {
+      const text = `
+      pipeline Test {
+        block CarsExtractor oftype HttpExtractor {
+          url: "https://gist.githubusercontent.com/noamross/e5d3e859aa0c794be10b/raw/b999fb4425b54c63cab088c0ce2c0d6ce961a563/cars.csv";
+        }
+
+        block CarsLoader oftype SQLiteLoader {
+          table: "Cars";
+          file: "./cars.sqlite";
+        }
+
+        pipe {
+          from: CarsExtractor;
+          to: CarsLoader;
+        }
+      }
+      `;
+
+      const parseResult = await parse(text);
+
+      const pipe: PipeDefinition = extractPipe(parseResult);
+
+      validatePipeDefinition(
+        pipe,
+        new ValidationContext(validationAcceptorMock),
+      );
+
+      expect(validationAcceptorMock).toHaveBeenCalledTimes(2);
+      expect(validationAcceptorMock).toHaveBeenNthCalledWith(
+        2,
+        'error',
+        `The output type "File" of HttpExtractor is incompatible with the input type "Table" of SQLiteLoader`,
+        expect.any(Object),
+      );
+    });
+  });
+
+  describe('chained pipe', () => {
+    // This test should succeed, because the error is thrown by langium during linking, not during validation!
+    it('should have no error even if pipe references non existing block', async () => {
+      const text = `
+      pipeline Test {
+        CarsExtractor -> CarsLoader;
+      }
+      `;
+
+      const parseResult = await parse(text);
+
+      const pipe: PipeDefinition = extractPipe(parseResult);
+
+      validatePipeDefinition(
+        pipe,
+        new ValidationContext(validationAcceptorMock),
+      );
+
+      expect(validationAcceptorMock).toHaveBeenCalledTimes(0);
+    });
+
+    it('should have no error even if pipe references block of non existing type', async () => {
+      const text = `
+      pipeline Test {
+        block UnknownOutput oftype UnknownOutputType {
+        }
+
+        block UnknownInput oftype UnknownInputType {
+        }
+        
+        UnknownOutput -> UnknownInput;
+      }
+      `;
+
+      const parseResult = await parse(text);
+
+      const pipe: PipeDefinition = extractPipe(parseResult);
+
+      validatePipeDefinition(
+        pipe,
+        new ValidationContext(validationAcceptorMock),
+      );
+
+      expect(validationAcceptorMock).toHaveBeenCalledTimes(0);
+    });
+
+    it('error on unsupported pipe between Blocktypes', async () => {
+      const text = `
+      pipeline Test {
+        block CarsExtractor oftype HttpExtractor {
+          url: "https://gist.githubusercontent.com/noamross/e5d3e859aa0c794be10b/raw/b999fb4425b54c63cab088c0ce2c0d6ce961a563/cars.csv";
+        }
+
+        block CarsLoader oftype SQLiteLoader {
+          table: "Cars";
+          file: "./cars.sqlite";
+        }
+        
+        CarsExtractor -> CarsLoader;
+      }
+      `;
```

Comment:
I wonder if we should consider pulling the models out as variables at the bottom of the file.
If we can give these variables good names, the tests might be more compact and easier to read.

Reply:
Or alternatively pull them out as separate `.jv` files and have them as test assets. This may ease migrating their syntax to new Jayvee versions once there is some kind of such migration mechanism in place.

Reply:
They have been extracted into own `.jv` files inside `test/assets/*` and are loaded using an utility function in `utils.ts`